motion_computation/src/motion_computation_node.cpp

```diff
@@ -61,4 +69,18 @@ namespace object{
     nh_.spin();
   }
 
-}//object
+  tf2::Transform MotionComputationNode::lookupECEFtoMapTransform()
+  {
+    tf2::Transform map_in_earth;
+    try
+    {
+      tf2::convert(tf_buffer_.lookupTransform("earth", "map", ros::Time(0)).transform, map_in_earth); //save to local copy of transform
+    }
+    catch (const tf2::TransformException &ex)
+    {
+      ROS_ERROR_STREAM("Could not lookup transform with exception: " << ex.what());
+    }
+    return map_in_earth;
```

Comment:
There is a possiblity the transform would not be available before. This is called. I think there is a tf2::waitForTransform method you could use to handle this

Reply:
looks like tf2 doesn't have it, and it was replaced by canTransform with timeout duration. Let me know if that is okay